### ISSUE TITLE
Fix S2372 FP: Add support for method invocations

### ISSUE THREAD

**Issue by gregory-paidis-sonarsource** [SonarSource/sonar-dotnet#8611]
### Description

If a property getter throws an exception of the allowed types via method Invocation, the rule raises.

### Repro steps

```csharp
        public int MyProperty
        {
            get
            {
                throw Method(); // Noncompliant FP
            }
        }

        private NotSupportedException Method() =>      
            new NotSupportedException(); // Belongs to the whitelisted exception types.
        
```

### Expected behavior

The rule should not raise.

### Actual behavior

The rule raises.
